substrate/client/network/statement/src/lib.rs

```diff
@@ -470,9 +516,56 @@ where
 
 			propagated_statements += to_send.len();
 
-			if !to_send.is_empty() {
-				log::trace!(target: LOG_TARGET, "Sending {} statements to {}", to_send.len(), who);
-				self.notification_service.send_sync_notification(who, to_send.encode());
+			let mut offset = 0;
+			while offset < to_send.len() {
+				// Try to send as many statements as possible in one notification
+				let chunk_size = to_send.len() - offset;
+				let chunk_end = offset + chunk_size;
+				let mut current_end = chunk_end;
+
+				loop {
+					let chunk = &to_send[offset..current_end];
+					let encoded = chunk.encode();
+
+					// If chunk fits, send it
+					if encoded.len() <= MAX_STATEMENT_NOTIFICATION_SIZE as usize {
```

Comment:
```suggestion
					if chunk.encoded_size() <= MAX_STATEMENT_NOTIFICATION_SIZE as usize {
```

`encoded_size()` is more performant and prevents allocations.